### ISSUE TITLE
Update Python packages for Python 3.x and pyspacer rollout

### ISSUE THREAD

**Issue by StephenChan** [beijbom/coralnet#185]
PyPI makes compatibility mostly easy to check, using the left sidebar. Checked on 2018/06:

Packages that depend on Django:

- django-guardian 1.4.9: Python 2.7, Python 3.6, Django 1.11; NOT Django 2.0
  - [Link](https://github.com/django-guardian/django-guardian/pull/537/commits/e98443503e885fec23a713a0633676fc9f674108): Django 2.0 support added in master branch, after the stable release.
- django-registration 2.4.1: Python 2.7, Python 3.6, Django 1.11; NOT Django 2.0
  - [Link](https://github.com/ubernostrum/django-registration/issues/134): Django 2.0 support is planned for django-registration 3.0.
- django-reversion 2.0.13: Python 2.7, Python 3.6, Django 1.11, Django 2.0
  - [Link](https://github.com/etianen/django-reversion/issues/717) confirming 3.6. [Link](https://github.com/etianen/django-reversion/issues/701) confirming 2.0
- django-storages 1.6.6: Python 2.7, Python 3.6, Django 1.11, Django 2.0
- easy-thumbnails 2.5.0: Python 2.7, Python 3.6, Django 1.11; NOT Django 2.0
  - "Support Django versions up to 1.11. Version 2.0 is in beta."

Packages that don't depend on Django:

- beautifulsoup4 4.6.0: Python 2.7, Python 3.6
  - Python 3.x actually, but presumably 3.6 is also okay.
- boto3 1.7.35: Python 2.7, Python 3.6
  - Per #82, we'll upgrade from boto to boto3. boto only goes up to Python 3.4.
- celery 4.2.0: Python 2.7, Python 3.6
- numpy 1.14.4: Python 2.7, Python 3.6
- pathlib2 2.3.2: Python 2.7, Python 3.6
  - This package is a shim for an import that only exists in 3.x, so it can be removed once we upgrade.
- Pillow 5.1.0: Python 2.7, Python 3.6
- psycopg2 2.7.4: Python 2.7, Python 3.6
- pytz 2016.3: Python 2.7, Python 3.6
- redis 2.10.6: Python 2.7, Python 3.6
- scikit-learn 0.19.1: Python 2.7, Python 3.6
- scipy 1.1.0: Python 2.7, Python 3.6
- six 1.12.0: Python 2.7, Python 3.6
- supervisor 3.3.4: Python 2.7; NOT Python 3.6
  - Python 3.x is supported in the master branch, and will be in supervisor 4.0. See Supervisor/supervisor#1060. Note that supervisor would run separately from the Django server, so Django can run on Py3 while this stays on Py2.

So to summarize:

- Update all our packages to the latest version, including boto -> boto3. They are all compatible with Python 2.7 and 3.6, except supervisor.
- supervisor must stay on Python 2.x for now, but that's fine.
- When we're safely running Django on Python 3.x, remove pathlib2.
- Keep Django at version 1.11 for now. We're still waiting on three packages before we can upgrade to Django 2.0.

**Comment by StephenChan** [comment]
As of PR #309, the package versions we have in requirements all work on Python 2.7 and 3.6. numpy 1.11.1 is troublesome to install on Windows though, and we want to update that to agree with pyspacer's numpy dependency anyway.

**Comment by StephenChan** [comment]
I forgot... it seems we also want Pillow 6.2.0 (currently 4.3.0) so that it matches pyspacer's dependency. That would make things a bit trickier. We have a few options regarding Pillow and django-andablog:

1. Keep django-andablog and upgrade everything 'properly':

   - Establish project as Python 3.x only
   - Upgrade to Django 2.0 (requires Python 3.x)
   - Upgrade django-andablog (requires Django 2.0)
   - Upgrade Pillow to 6.2.0 (requires latest django-andablog)

2. After installing django-andablog in a particular environment, monkey-patch its Pillow requirement so that we can upgrade to Pillow 6.2.0 while keeping the older andablog version.

3. Remove the django-andablog package and just copy the parts that we use, as discussed in issue #249's comments.

For 1, I'd prefer not to do Django 2.0 porting in the same round as Python 3.x + pyspacer. For 2, monkey-patching requirements is pretty unappealing especially if we rely on Travis. And for 3... well, I'd prefer not to mess with model migrations at this stage, but I guess this case shouldn't be too bad. So I think 3 is the least-unappealing option.

**Comment by beijbom** [comment]
A fourth option is to relax the PILLOW requirements on spacer. Let me check if that is possible. I think it is.

**Comment by StephenChan** [comment]
Okay, that'll help, thanks. Just so you know though, Pillow 4.3.0 makes me get the libjpeg9 result for the image reading test. Might not matter in the end though, since I plan to do Django 2.0 shortly after the pyspacer rollout, and updating Pillow would be easy after that.

**Comment by beijbom** [comment]
Ok. See comment in https://github.com/beijbom/pyspacer/pull/10. The pillow version doesn't really matter, it's the `libjpeg` version. Ffor Coralnet, the only time this will matter is what is deployed in the spacer cluster which will be the docker build.

**Comment by StephenChan** [comment]
This should be done now. To recap the recent changes:

- https://github.com/beijbom/pyspacer/pull/10 ensured that pyspacer is compatible with our existing Pillow and boto versions.
- PR #309 removed the scipy and scikit-learn dependencies.
- PR #312 updated the numpy dependency to match pyspacer.
